### PR TITLE
✅  🏗 ♻️  Remove puppeteer from e2e tests

### PR DESCRIPTION
**Pull request by rileyajones**
Unfortunately puppeteer still does not support all of the browsers we would like and having support for running the tests in two different frameworks leads to messy code. Puppeteer is only used by e2e tests when they are run locally with the command line argument `engine` set to `PUPPETEER` so it should be safe to remove.

### CODE REVIEW

**Review by rsimha — COMMENTED** [review]
Code changes LGTM. One important comment below about the lockfile.

**Note to self:** Expand `build-system/pr-check/npm-checks.js` to also run on sub-packages during CI builds.
**Edit:** Sent out a PR that does this: #33880

File: build-system/tasks/e2e/package-lock.json [hunk 1]
```diff
@@ -1,8 +1,1413 @@
 {
   "name": "amp-e2e",
   "version": "0.1.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply build-system/tasks/e2e/package-lock.json]: I had a feeling there would be something like this in here which is why it's currently set as draft. As for `npm` versions my laptop has been crashing a lot lately and when it does I keep having to change the node version (finally set the `nvm default` alias).

File: build-system/tasks/e2e/describes-e2e.js [hunk 2]
```diff
@@ -462,23 +428,12 @@ function describeEnv(factory) {
      * @return {Set<string>}
      */
     function getAllowedBrowsers() {
-      const {engine, browsers} = getConfig();
+      const {browsers} = getConfig();
```
Comment (estherkim) [inline build-system/tasks/e2e/describes-e2e.js:431]: Can you remove the `engine` flag in the e2e task as well?

Comment (rileyajones) [reply build-system/tasks/e2e/describes-e2e.js:431]: Great catch.

**Review by rsimha — APPROVED** [review]
LGTM. Nice cleanup!

**Review by estherkim — APPROVED** [review]
Thanks Riley!

**Pull request merged**